__tests__/Ship.test.js

```diff
@@ -0,0 +1,54 @@
+/* globals describe it expect */
+const Ship = require("../src/Ship.js");
+const Port = require("../src/Port.js");
+const Itinerary = require("../src/Itinerary.js");
+
+describe("Ship", () => {
+  let ship;
+  let dover;
+  let calais;
+  let itinerary;
+
+  beforeEach(() => {
+    dover = new Port("Dover");
+    calais = new Port("Calais");
+    itinerary = new Itinerary([dover, calais]);
+    ship = new Ship(itinerary);
+  });
+
+  describe("Ship constructor", () => {
+    xit("can be instantiated", () => {
+      expect(Ship).toBeInstanceOf(Object);
+    });
+
+    xit("it has a starting port", () => {
+      expect(ship.currentPort).toBe(dover);
+    });
+
+    xit("can set sail", () => {
+      ship.setSail();
+
+      expect(ship.currentPort).toBeFalsy();
+      expect(dover.ships).not.toContain(ship);
```

Comment:
This doesnt really work. Youre looking to see if you can assert whether or not the current port has been left, that the number of upcoming ports has been reduced by 1 (so using your test data you have here this will mean just 1 remaining), and that its name will be Calais.
You can do this with these assertions:
```
expect(ship.remainingPort.length).toBe(1);
expect(ship.remainingPort[0].name).toBe("Calais");
```